Modernize to Java 17 switch expression.

deepika-u opened the pull request:
Update legacy switch block with Java 17 switch expression in org.eclipse.ui.workbench.texteditor package.

deepika-u committed:
Modernize to Java 17 switch expression.Added default case.

iloveeclipse commented on bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/internal/texteditor/rulers/RulerColumnPlacement.java:
```diff
@@ -67,18 +67,14 @@ private Set<RulerColumnPlacementConstraint> readIds(IConfigurationElement[] chil
+				case BEFORE -> true;
+				default -> {
+					childHelper.fail(RulerColumnMessages.RulerColumnPlacement_illegal_child_msg);
+					yield false; // Required because default must return a value.
```
This looks wrong. Is this autogenerated or manually changed?
Original code would continue for loop without adding anything to constraints.

deepika-u replied:
I had a doubt only in this context. So here it cannot be written in newer syntax does it mean?

iloveeclipse replied:
No idea, have no time for this.

deepika-u replied:
can this loop be like this to achieve the loop?
```
boolean before = switch (name) {
    case AFTER -> false;
    case BEFORE -> true;
    default -> {
        childHelper.fail(RulerColumnMessages.RulerColumnPlacement_illegal_child_msg);
        yield false;
    }
};
continue;
```

iloveeclipse replied:
No!

deepika-u replied:
Let me check on it further.

deepika-u replied:
Reverting this to original switch, thanks for the suggestion.

iloveeclipse commented:
I'm not sure the change improves readability.
Is this PR manually crafted of is the result of automated quick fix?

deepika-u commented:
> I'm not sure the change improves readability. Is this PR manually crafted of is the result of automated quick fix?

I have did it manually.

iloveeclipse commented:
> I have did it manually

Why not with a cleanup? Or there is nothing available?

deepika-u commented:
> > I have did it manually
> 
> Why not with a cleanup? Or there is nothing available?

Any specific context of "cleanup" in this code are you referring to?

iloveeclipse reviewed (changes requested):
You can perform changes via Source->Cleanup... and the cleanup of your choice.
![image](https://github.com/user-attachments/assets/50b0b9cc-06b3-4ac2-b0f9-0e16827546fb)

The proposed change breaks code.

deepika-u commented:
> You can perform changes via Source->Cleanup... and the cleanup of your choice. 
> 

Thanks for the suggestion, even via this that file is not getting picked and updated. So i have reverted that file change(switch to original code). 
Can you review now please?

iloveeclipse commented:
> even via this that file is not getting picked and updated

That is the sign that the quick fix doesn't have an automated solution for conversion (yet).

> Can you review now please?

The new change shows some new warning, shows files with whitespace only changes. That shouldn't happen.

I only briefly checked **original PR** whether the proposed change make sense, found immediately a problem and beside the problem nothing interesting in the change itself. Approving a PR means I agree with the change, which is not the case here. 

I would propose to close this PR, and if you are eager to apply this particular refactoring on this particular bundle, do this with the automated cleanup **and only with that**.

For the new PR please find someone who is interested in such kind of reviews and let review it. I'm definitely not interested in this particular change.

akurtakov commented:
What is the status of this one?

deepika-u commented:
@akurtakov 
Did already addressed all the review comments regarding the code changes.
But the above code changes are from automated clean up and additional manual changes(which were missed by automated clean up tool) as well.

Updating the automated clean up - i agree needs to be improved but not sure how and who does it exactly(may be via JDT or JDT ui) which can be taken up separately.

So when you get some free time, can you review these changes please?

akurtakov commented:
@iloveeclipse Your change request is pending here. Is the PR good according to you?

iloveeclipse commented:
> @iloveeclipse Your change request is pending here. Is the PR good according to you?

It is not mine and I've answered already above: https://github.com/eclipse-platform/eclipse.platform.ui/pull/2816#issuecomment-2684899239

akurtakov commented:
> > @iloveeclipse Your change request is pending here. Is the PR good according to you?
> 
> It is not mine and I've answered already above: [#2816 (comment)](https://github.com/eclipse-platform/eclipse.platform.ui/pull/2816#issuecomment-2684899239)

![image](https://github.com/user-attachments/assets/506bb5fb-08e8-4c18-9d47-e9303823ab74)

iloveeclipse commented:
I prefer to close this PR, I don't see any value in this change and the time ro properly review is not worth it.
